test/plugins/cxxrandom.lua

```diff
@@ -0,0 +1,75 @@
+local rng = require('plugins.cxxrandom')
+
+function test.cxxrandom_distributions()
+    rng.normal_distribution:new(0,5)
+    rng.real_distribution:new(-1,1)
+    rng.int_distribution:new(-20,20)
+    rng.bool_distribution:new(0.00000000001)
+    rng.num_sequence:new(-1000,1000)
+    -- no errors, no problem
+end
+
+--[[
+The below tests pass with their given seeds, if they begin failing
+for a given platform, or all around, new seeds should be found.
+
+Note: these tests which assert RNG, are mere sanity checks
+to ensure things haven't been severely broken by any changes
+]]
+
+function test.cxxrandom_seed()
+    local nd = rng.normal_distribution:new(0,500000)
+    local e1 = rng.MakeNewEngine(1)
+    local e2 = rng.MakeNewEngine(1)
+    local e3 = rng.MakeNewEngine(2)
+    local g1 = rng.crng:new(e1, true, nd)
+    local g2 = rng.crng:new(e2, true, nd)
+    local g3 = rng.crng:new(e3, true, nd)
+    expect.eq(g1:next(), g2:next())
+    expect.ne(g1:next(), g3:next())
```

Comment:
I think this would pass even if all three engines started with the same seed. This compares the *second* output of `g1` with the *first* output of `g3`. I think what is intended here is that we ensure the first output of the two engines are different. Storing the first output of `g1` in a local and using that in the comparisons would fix it, I think.

Reply:
Oops! Nice catch. I wrote these tests perhaps a bit too quickly